src/schema.ts

```diff
@@ -174,11 +174,14 @@ export class Schema {
       }
       // if the cardinality for the timeUnit is not cached, calculate it
       if (!fieldSchema.timeStats[encQ.timeUnit as string]) {
-        fieldSchema.timeStats[encQ.timeUnit as string] = timeSummary(encQ.timeUnit as TimeUnit, fieldSchema.stats);
+        fieldSchema.timeStats[encQ.timeUnit as string] = timeSummary(encQ.timeUnit as TimeUnit, fieldSchema.stats, excludeInvalid);
```

Comment:
As said [here](https://github.com/uwdata/compassql/pull/224/files#r74876060), you should always cache the full distinct and subtract `invalidCount()` if the flag is enabled.